addons/mail/static/src/owl/components/composer_text_input/composer_text_input.js

```diff
@@ -161,14 +274,84 @@ class ComposerTextInput extends Component {
      */
     _onKeydownTextarea(ev) {
         switch (ev.key) {
+            // UP, DOWN, TAB: prevent moving cursor if navigation in mention propositions
+            case 'ArrowUp':
+            case 'ArrowDown':
+            case 'Tab':
+                if (this.state.isMentionsOpen) {
+                    ev.preventDefault();
+                }
+                break;
+            // ENTER: submit the message only if the dropdown mention proposition is not displayed
             case 'Enter':
-                this._onKeydownTextareaEnter(ev);
+                if (this.state.isMentionsOpen) {
+                    ev.preventDefault();
+                } else {
+                    this._onKeydownTextareaEnter(ev);
+                }
+                break;
+        }
+    }
+
+    propositionNavigation(key) {
+        const active_id = this.state.active_id;
+        let active = this.state.mentionSuggestions[0].find(function(item) {
+            return item.id === active_id;
+        });
+        const selectionIndex = this.state.mentionSuggestions[0].indexOf(active);
+        if (key === 'Enter') {
+            // selecting proposition
+            this._insertMentionSuggestion(active);
+        } else {
+            // navigation in propositions
+            if (key === 'ArrowDown' && selectionIndex !== this.state.mentionSuggestions[0].length - 1) {
+                active = this.state.mentionSuggestions[0][selectionIndex + 1];
+                this.state.active_id = active.id;
+            } else if (key === 'ArrowUp' && selectionIndex !== 0) {
+                active = this.state.mentionSuggestions[0][selectionIndex - 1];
+                this.state.active_id = active.id;
+            } else if (key === 'Tab') {
+                if (selectionIndex === this.state.mentionSuggestions[0].length - 1) {
+                    active = this.state.mentionSuggestions[0][0];
+                }
+                else {
+                    active = this.state.mentionSuggestions[0][selectionIndex + 1];
+                }
+                this.state.active_id = active.id;
+            }
+        }
+    }
+
+    _onKeyupTextarea(ev) {
+        switch (ev.key) {
+            // ESCAPED KEYS: do nothing
```

Comment:
Why do nothing for those? Naively for me it would seem convenient to navigate with them too. Also probably want to include the home button in the mix

Reply:
Done